packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -621,6 +623,52 @@ export class SplitV2Client extends SplitV2Transactions {
   async owner(split: Address): Promise<Address> {
     return this._getSplitV2Contract(split).read.owner()
   }
+
+  async getSplitMetadata(splitAddress: Address): Promise<{ split: SplitV2 }> {
+    this._requirePublicClient()
+
+    const [updateLogs, createLogs, owner] = await Promise.all([
+      this._publicClient?.getLogs({
+        address: splitAddress,
+        event: splitUpdatedEvent,
+        strict: true,
+      }),
+      this._publicClient?.getLogs({
+        event: splitCreatedEvent,
```

Comment:
this event comes from the factory, right? should we be setting the factory as the `address`? I see your filtering on the args, just not sure if filtering on the `address` is more efficient/better for any reason.

Reply:
yeah, adding the factory as the address makes sense